### PR TITLE
Raytracing device extensions

### PR DESCRIPTION
**Pull request by Erfan-Ahmadi**
## Description
Added Extensions and internal helper functions to add extensions

## Testing 
It was tested with multiple extensions and features

### CODE REVIEW

File: include/nbl/video/IPhysicalDevice.h [hunk 1]
```diff
@@ -136,13 +142,33 @@ class IPhysicalDevice : public core::Interface, public core::Unmovable
...
+    {
+        assert(false); // Invalid code path
+        return false;
+    }
 
+    virtual bool isRayQuerySupported() const
+    {
+        assert(false); // Invalid code path
+        return false;
+    }
```
Comment (devshgraphicsprogramming) [inline include/nbl/video/IPhysicalDevice.h:162]: I would have thought these would correllate to the booleans?

P.S. Dont assert here, the user is only asking for availability not actually using the extensions

Comment (Erfan-Ahmadi) [reply include/nbl/video/IPhysicalDevice.h:162]: Understood, I just followed what `isSwapchainSupported` did
I guess I should remove the assert from that function as well?

Comment (devshgraphicsprogramming) [reply include/nbl/video/IPhysicalDevice.h]: yes please

File: include/nbl/video/IPhysicalDevice.h [hunk 2]
```diff
@@ -113,7 +113,13 @@ class IPhysicalDevice : public core::Interface, public core::Unmovable
     virtual IDebugCallback* getDebugCallback() = 0;
 
     virtual bool isSwapchainSupported() const = 0;
+        
+    virtual bool isRayTracingPipelineSupported() const = 0;
 
+    virtual bool isAccelerationStructuresSupported() const = 0;
+
+    virtual bool isRayQuerySupported() const = 0;
```
Comment (devshgraphicsprogramming) [inline include/nbl/video/IPhysicalDevice.h:121]: I'm a bit concerned about redundancy here, lots and lots of code to find out what's already present in 
```cpp
    const SLimits& getLimits() const { return m_limits; }
    const SFeatures& getFeatures() const { return m_features; }
```
anyway

Comment (Erfan-Ahmadi) [reply include/nbl/video/IPhysicalDevice.h:121]: Do you want me to remove these functions?  @devshgraphicsprogramming

Comment (devshgraphicsprogramming) [reply include/nbl/video/IPhysicalDevice.h:121]: yes, would make most sense.

**Comment by devshgraphicsprogramming** [comment]
@Erfan-Ahmadi have @Crisspl review instead of me

**Comment by Erfan-Ahmadi** [comment]
Seems like It's already being used and integrated in vulkan_backend, deferring review for that branch, closing here and deleting this useless branch

**Pull request closed without merging**